config/production.yaml.example

```diff
@@ -33,6 +33,17 @@ redis:
   auth: null
   db: 0
 
+ldap:
+  enable: true
+  # Disallow non-ldap users (it also disables root!)
+  ldap_only: true
+  url: ldap://localhost:389/dc=example,dc=com
+  bind_dn: cn=admin,dc=example,dc=com
+  bind_password: adminPass
+  base: dc=example,dc=com
+  mail_entry: "mail"
+  user_filter: "(|(email=%username%)(uid=%username%))"
+
```

Comment:
You may want to add a setting to prevent connection failure when you use ldaps and the LDAP server has a self-signed certificate (for ex.).

I didn't use the same module than you to connect to LDAP, but on [ep_mypads](https://framagit.org/framasoft/Etherpad/ep_mypads/), I had to add this optional setting in the LDAP settings:

```
    "tlsOptions": {
        "rejectUnauthorized": false
    },
```

If you want to easily test your code against a self-signed certificated ldaps, you can use this docker image: https://github.com/rroemhild/docker-test-openldap

Reply:
Good catch @ldidry I added that option.

I used inspiration from etherpad-lite to write this PR, so it’s definitely the same node module ldapjs that is used in both cases :)

(And thanks for pointing me to ep_mypads module, it seems great and I didn’t know about!)